json_serializable/lib/builder.dart

```diff
@@ -23,8 +23,12 @@ import 'src/json_part_builder.dart';
 /// Not meant to be invoked by hand-authored code.
 Builder jsonSerializable(BuilderOptions options) {
   try {
-    final config = JsonSerializable.fromJson(options.config);
-    return jsonPartBuilder(config: config);
+    final configOptions = Map<String, dynamic>.of(options.config);
+    final ignoreForFile = Set<String>.from(
+      configOptions.remove('ignore_for_file') as List ?? <String>[],
```

Comment:
One could argue we could/should validate these entries to be something like `[a-z_]+` or some such.

Thoughts, @natebosch ?